File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown,yellow):
    # 테두리의 둘래 = 갈색 격자의 수
    # 카펫 가로 길이 >= 세로길이
    # 꼭짓점을 뺀 둘래를 반으로 나누고 x,y를 임의로 정한다
    # x*y == 노란색인 x,y를 찾는다
    # x,y는 1이상 이어야 한다. 
    # 더큰 것을 먼저 반환
    # 반환(more+2, less+2)
    dule = int(((brown-4)/2))
    for x,y in [ (i, dule - i) for i in range(1,dule)]:
        #print(x,y)
        if x*y == yellow:
            return [max([x,y])+2, min(x,y)+2]

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public int[] solution(int brown, int yellow) {
        /*
        합의 제곱근까지 약수 구해서 
        가로 세로 가정해서 (가로-2) * (세로-2) = yellow인지 판단하기
        */
        int sum = brown + yellow;
        int root = (int)Math.sqrt(sum);
        
        //log(root);
        for(int height = 3; height <= root; height++){
            if(sum % height == 0){
                int width = sum / height;
                int expected = (height - 2) * (width - 2);
                if(expected == yellow){
                    return new int[]{width, height};
                }
            }
        }
        return new int[]{};
    }
}